src/fiskaltrust.Launcher/Commands/Common.cs

```diff
@@ -229,42 +239,88 @@ public static class CommonHandler
+                            {
+                                Log.Warning("Failed to change owner of the service directory.");
+                            }
+
+                            var chmodResult = await ProcessHelper.RunProcess("chmod", new[] { "774", serviceDirectory }, LogEventLevel.Debug);
+                            if (chmodResult.exitCode != 0)
+                            {
+                                Log.Warning("Failed to change permissions of the service directory.");
+                            }
+                        }
+                        else
+                        {
+                            Log.Warning("Service user name is not set. Owner of the service directory will not be changed.");
+                        }
+                    }
+                    else
+                    {
+                        Log.Debug("Changing owner and permissions is skipped on non-Unix operating systems.");
+                    }
+                }
+            }
+            catch (UnauthorizedAccessException e)
+            {
+                // will exit with non-zero exit code later.
+                Log.Fatal(e, "Access to the path '{ServiceDirectory}' is denied. Please run the application with sufficient permissions.", serviceDirectory);
+            }
+        }
+
         public static async Task<ECDiffieHellman> LoadCurve(Guid cashboxId, string accessToken, string serviceFolder, bool useOffline = false, bool dryRun = false, bool useFallback = false)
         {
             Log.Verbose("Loading Curve.");
             var dataProtector = DataProtectionExtensions.Create(accessToken, useFallback: useFallback).CreateProtector(CashBoxConfigurationExt.DATA_PROTECTION_DATA_PURPOSE);
             var clientEcdhPath = Path.Combine(serviceFolder, $"client-{cashboxId}.ecdh");
 
-            if (File.Exists(clientEcdhPath))
-            {
-                return ECDiffieHellmanExt.Deserialize(dataProtector.Unprotect(await File.ReadAllTextAsync(clientEcdhPath)));
-            }
-            else
+            try
             {
-                const string offlineClientEcdhPath = "/client.ecdh";
-                ECDiffieHellman clientEcdh;
-
-                if (!dryRun && useOffline && File.Exists(offlineClientEcdhPath))
+                if (File.Exists(clientEcdhPath))
                 {
-                    clientEcdh = ECDiffieHellmanExt.Deserialize(await File.ReadAllTextAsync(offlineClientEcdhPath));
+                    var protectedData = await File.ReadAllTextAsync(clientEcdhPath);
                     try
                     {
-                        File.Delete(offlineClientEcdhPath);
+                        return ECDiffieHellmanExt.Deserialize(dataProtector.Unprotect(protectedData));
+                    }
+                    catch (Exception ex)
+                    {
+                        Log.Warning($"Could not decrypt ECDH curve, regenerating a new one. Error: {ex.Message}");
```

Comment:
There is an overload for the logging methods that have the exception as the first parameter.
```suggestion
                        Log.Warning(ex, $"Could not decrypt ECDH curve, regenerating a new one.");
```